### ISSUE TITLE
audiobooks metadata 

### ISSUE THREAD

**Issue by icf20** [MediaBrowser/Emby#3518]
emby version Version 4.0.1.0
the issue well there is no way to get the cover of the audio book even if from paper because there is no ISBN 

example of audio book ISBN:978-0140442106

![image](https://user-images.githubusercontent.com/17784575/51792787-cd154b00-21b6-11e9-8480-adf1dc3e746e.png)

is true that musicbrainz have a profile for him https://musicbrainz.org/artist/d787fd08-ece5-4b64-8670-7208765601e1 i would like emby to grab the book cover via ISBN search instead of having to download everything by hand 

so my first suggestion is to have the image downloaded automatically using ISBN 

second suggestion is to have the category for the audiobook instead of genre, since audiobook are just book in audio format, one of the system for categorizing books is melvil dewey system in case of the book above is 188 ( Philosophy and Psychology > Ancient, medieval and eastern philosophy > Stoic) https://www.librarything.com/mds/188 

to get there you only need to search for ISBN and u get this https://www.librarything.com/work/16622 
**DDC/MDS | 188** the same page also show the library of congress **LC Classification | BJ214 .S4** if anyone wants that instead of dewey

**Comment by LukePulverenti** [comment]
We currently don't download audio book images from the internet. They either have to be embedded into the files, or in your media folders.

If we allow you to enter ISBN, what sites can the images be downloaded from?

**Comment by koelner21** [comment]
Hello,

Do you mean something like this?

https://b-ok.org/

Mit freundlichen Grüßen

David Dieter

**Comment by LukePulverenti** [comment]
> Hello, Do you mean something like this? https://b-ok.org/ Mit freundlichen Grüßen David Dieter
> […](#)
> Am 22.06.2019 um 19:39 schrieb Luke ***@***.***>: We currently don't download audio book images from the internet. They either have to be embedded into the files, or in your media folders. If we allow you to enter ISBN, what sites can the images be downloaded from? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or mute the thread.

Yes although I don't see any API information for third part apps to use their data.

**Comment by koelner21** [comment]
This side will help you. There you can find APIs for many categorys

https://www.programmableweb.com/category/ebooks/api

Mit freundlichen Grüßen

David

**Comment by LukePulverenti** [comment]
What info would you want to pull from LibraryThing? Their descriptions are very short, and they don't have genres. The Tags are probably the most useful bit of info they have.

**Comment by LukePulverenti** [comment]
> This side will help you. There you can find APIs for many categorys https://www.programmableweb.com/category/ebooks/api Mit freundlichen Grüßen David
> […](#)
> Am 29.06.2019 um 18:26 schrieb Luke ***@***.***>: Hello, Do you mean something like this? https://b-ok.org/ Mit freundlichen Grüßen David Dieter … Am 22.06.2019 um 19:39 schrieb Luke @.***>: We currently don't download audio book images from the internet. They either have to be embedded into the files, or in your media folders. If we allow you to enter ISBN, what sites can the images be downloaded from? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or mute the thread. Yes although I don't see any API information for third part apps to use their data. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.

@koelner21 do you have experience with any of those? which do you think is the best?

**Comment by ProphetLamb** [comment]
The [Goodreads API](https://www.goodreads.com/api) allows access to book metadata.

- [book.isbn_to_id](https://www.goodreads.com/api/index#book.isbn_to_id)   —   Get Goodreads book IDs given ISBNs.
- [book.id_to_work_id](https://www.goodreads.com/api/index#book.id_to_work_id)   —   Get Goodreads work IDs given Goodreads book IDs.
- [book.review_counts](https://www.goodreads.com/api/index#book.review_counts)   —   Get review statistics given a list of ISBNs.
- [book.show](https://www.goodreads.com/api/index#book.show)   —   Get the reviews for a book given a Goodreads book id.
- [book.show_by_isbn](https://www.goodreads.com/api/index#book.show_by_isbn)   —   Get the reviews for a book given an ISBN

Tho the user would have to configure an API key. I am toying around in insomnia right now.
Edit: It seems they are retiering thier service. 
On another look I found [openlibary](https://openlibrary.org/dev/docs/api/books) I looks pretty good.
Really thier API does not require a key and seems on point. Tho you'll need the [path params](https://insomnia.rest/plugins/insomnia-plugin-path-parameters) plugin in insomnia.

For the query `https://openlibrary.org/isbn/:isbn.json` with `isbn = 9780441569595` (Neuromancer by William Gibson)  we get the following response

```json
{
    "publishers": [
        "Berkley Publishing Group"
    ],
    "contributors": [
        {
            "role": "Cover artist",
            "name": "Richard Berry"
        }
    ],
    "source_records": [
        "marc:marc_loc_updates/v40.i08.records.utf8:3309677:976"
    ],
    "title": "Neuromancer",
    "number_of_pages": 271,
    "created": {
        "type": "/type/datetime",
        "value": "2010-05-08T14:51:38.658100"
    },
    "physical_format": "Paperback",
    "isbn_10": [
        "0441569595"
    ],
    "publish_date": "1986",
    "key": "/books/OL24226054M",
    "last_modified": {
        "type": "/type/datetime",
        "value": "2020-08-22T06:33:58.518250"
    },
    "latest_revision": 5,
    "publish_places": [
        "New York, NY"
    ],
    "works": [
        {
            "key": "/works/OL27258W"
        }
    ],
    "type": {
        "key": "/type/edition"
    },
    "revision": 5
}
```

As you can see, we get two keys one `/books/OL24226054M` and another `/works/OL27258W` with which we can gather further data from the API. Since the API is exclusive to book we wont actually get any information about the narrator.

**Comment by ProphetLamb** [comment]
Upon further research, I found a project that actually uses the elusive audible API. Tho I haven't been able to get it to work property just yet. [AudibleAPI](https://github.com/rmcrackan/AudibleApi)
Looking at the code, it seems rather resource intense. Not the best for emby, but it might just be the most complete tool to grab audiobook metadata, if we can get it to work

**Comment by LukePulverenti** [comment]
There's now an OpenLibrary plugin in the Emby plugin catalog, and it supports ISBN.
